tooling/minimize/src/function.rs

```diff
@@ -107,25 +107,87 @@ impl<'cx, 'tcx> FnCtxt<'cx, 'tcx> {
             self.locals.insert(*local_name, ty);
         }
 
+        let mut args = List::default();
+        let mut init_statements: Vec<Statement> = Vec::new();
+        let mut spread_assignments = Vec::new();
+
+        // spread_arg semantics
+        if let Some(spread_local) = self.body.spread_arg {
+            let spread_local_idx = spread_local.as_usize();
+            let spread_local_name = self.local_name_map[&spread_local];
+
+            let spread_decl = &self.body.local_decls[spread_local];
+            let (tuple_fields, span) = if let rs::TyKind::Tuple(fields) = spread_decl.ty.kind() {
+                (fields, spread_decl.source_info.span)
+            } else {
+                panic!(
+                    "spread_arg local {spread_local:?} is not a tuple, has type: {:?}",
+                    spread_decl.ty
+                );
+            };
+
+            // add all the args before the spread arg
+            for i in 1..spread_local_idx {
+                args.push(LocalName(Name::from_internal(i as _)));
+            }
```

Comment:
Instead of three separate loops, it would be much nicer to make this a single loop with an `if` inside that treats the one special local differently.